cephfs/userperm.go

```diff
@@ -0,0 +1,73 @@
+package cephfs
+
+/*
+#cgo LDFLAGS: -lcephfs
+#cgo CPPFLAGS: -D_FILE_OFFSET_BITS=64
+#include <stdlib.h>
+#include <cephfs/libcephfs.h>
+*/
+import "C"
+
+import (
+	"runtime"
+	"unsafe"
+)
+
+// UserPerm types may be used to get or change the credentials used by the
+// connection or some operations.
+type UserPerm struct {
+	userPerm *C.UserPerm
+
+	// cache create-time params
+	managed bool // if set, the userPerm was created by go-ceph
```

Comment:
Is this really needed?

Reply:
So if you read some of the doc comments in ceph's src/include/cephfs/libcephfs.h there are a bunch of caveats around the dealing with this type when its either allocated by the caller vs fetching the default one off of the mount object. There's also some related tricky stuff related to the gidlist. Because of all this I wanted to be very safe here and make sure we knew if the item pointed to by userPerm was allocated in our lib vs. by ceph. Even though it may not be strictly needed now I wanted to leave a breadcrumb here that clearly marked what kind of C.UserPerm the go type was managing.

In addition I didn't want users of our library to have to distinguish between the two states so that they could treat them the same from the POV of their Go code.

Reply:
Ah, I was wondering about why a user would allocate a `UserPerm` without going through `NewUserPerm()`. If it is about the allocation from libcephfs or Go, it makes more sense to me.